plugins/pulp_docker/plugins/token_util.py

```diff
@@ -82,12 +82,12 @@ def parse_401_response_headers(response_headers):
     """
     auth_header = response_headers.get('www-authenticate')
     if auth_header is None:
-        raise IOError("401 responses are expected to conatin authentication information")
+        raise IOError("401 responses are expected to contain authentication information")
     auth_header = auth_header[len("Bearer "):]
 
     # The remaining string consists of comma seperated key=value pairs
     auth_dict = {}
-    for key, value in (item.split('=') for item in auth_header.split(',')):
+    for key, value in (item.split('=') for item in auth_header.split('",')):
```

Comment:
Note that RFC 7235 explicitly updates RFC 2617. That said, we could make an argument that we only expect to see oath2's "Bearer" scheme in this header value, and that following convention, we expect all values to be quoted strings.

That's fine with me, and likely will work for all known use cases, but I do think we would be depending on convention. By my reading of the RFCs, it would be legal even for an oauth2 challenge to include an ``auth-param`` whose value is of type ``token``, which does not get quoted.

If there is something that can parse this for us, that would be a big improvement, but I'm ok with keeping this limited implementation if you feel confident it will meet all of our use cases.